### DISCUSSION TITLE
[@types/p5] v1.7.1 - Contains 5 typescript compilation errors

### DISCUSSION THREAD

**Discussion by christopherball** [DefinitelyTyped/DefinitelyTyped#67288] [Issues with a @types package] [answered]
I'm somewhat new to troubleshooting this sort of thing, but it appears there are 5 compilation issues with @types/p5:

![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/5790394/19d1032e-9c3a-4f15-bc51-4b14913201a8)

For the 2 vector errors, and the 1 SoundFile error, simply prepending "p5." in front resolves their issues.  For the 2 float-related errors, it appears to be a duplicate definition issue, so simply deleting one of them resolves both errors.  Can someone with more experience confirm these fixes are valid for said 5 errors, and if so, merge them into the mainline?  As of right now, I have no choice but to work off of my manual changes described above to get clean typescript compilations...Cheers!

![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/5790394/22810136-42c6-4b6f-9d70-9afdd2e68369)

[a comment by an automated account is not shown]

**Comment by jakebailey** [comment]
This package compiles for me; if you are working within the DT repo, did you forget to `pnpm install` (with requisite filters if you don't want to install all 9000 packages)?

**Reply by jakebailey** [reply]
Oh, huh; this file is not loaded in a regular `tsc` run; it's present in `OTEHR_FILES.txt` and isn't referenced. That sounds like a mistake.

**Reply by jakebailey** [reply]
This package says it's generated, but provides no info on how it's generated or who's supposed to fix it...

**Reply by christopherball** [reply]
That definitely sounds like the problem, if this package isn't actually vetted with a proper `tsc` run...I am simply using this specific package in a minimal starter template and observed `tsc` chokes on it.  For example, if you do a quick `npm install` and `npm start` on https://github.com/Gaweph/p5-typescript-starter.  It works fine as the version of @types/p5 mentioned in the package.json is older.  If you upgrade those values to the latest version, the above compilation errors result until I manually fix them.  E.g.:

```  "devDependencies": {
    "@types/p5": "^1.7.1",
    "browser-sync": "^2.29.3",
    "npm-run-all": "^4.1.5",
    "typescript": "^5.2.2"

**Reply by jakebailey** [reply]
> That definitely sounds like the problem, if this package isn't actually vetted with a proper `tsc` run...

It is, but the `tsconfig.json` doesn't include `globals.d.ts` and no test includes them either.

> It works fine as the version of @types/p5 mentioned in the package.json is older.

To be clear, you're saying that it used to work in some older `@types/p5` but now does not work? Do you have a pair of versions that work / don't work so we can diff the package?

**Reply by christopherball** [reply]
The proper includes are all there (edit: in the example project I mentioned, apparently not in this project space per your comments) in that same projects `global.d.ts` file sitting in the root.

```
import * as p5Global from 'p5/global' 
import module = require('p5');
export = module;
export as namespace p5;
```

> To be clear, you're saying that it used to work in some older @types/p5 but now does not work? Do you have a pair of versions that work / don't work so we can diff the package?

The package I referenced, if you make no changes at all, compiles cleanly, so at a minimum, the diff between version 1.3.3 and 1.7.1 would be a starting place.

**Reply by jakebailey** [reply]
That's... a wide range 😄 1.3 was last published like 2 years ago. But yeah, that user code references `global`, but the tests on DT do not. Definitely something to fix.

**Reply by christopherball** [reply]
While I'd be happy to help narrow the version gap down further since I have a vested interest in this being fixed, I'm otherwise not likely to be able to easily address the broader issue with DT to get this resolved.  Is this something you are able to commit to as the maintainer of this space :)?

**Reply by jakebailey** [reply]
To be clear I'm not a maintainer of the p5 types and know nothing about them besides what I am reading as of this conversation; I'm just on the TS team and help maintain DT as a whole.

What I believe happened was that https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65202 fixed your problem, but the types are generated, so the change was reverted in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66294.

@asukaminato0721

**Reply by asukaminato0721** [reply]
that double `FLOAT` is fixed by https://github.com/processing/p5.js/pull/6330 , other is fixing.

**Reply by christopherball** [reply]
That merge from 3 months ago did not fix the duplicate float declaration issue above or I wouldn't be seeing it in the latest v1.7.1 code.  I just checked and see [merge 6330](https://github.com/processing/p5.js/pull/6330) is already showing up under `@types/p5/src/core/constants.d.ts`, whereas all of the 5 issues reside within `@types/p5/global.d.ts`.

> ...other is fixing

Sounds like you're fixing the others as well, so thank you @asukaminato0721  - If you wouldn't mind dropping a link here for said fixes so that I can check back on their status that'd be appreciated.

**Reply by asukaminato0721** [reply, marked as answer]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67308

**Reply by christopherball** [reply]
v1.7.2 builds clean, thank you!!  どもありがとうございます！
